vcs/remote.go

```diff
@@ -23,5 +23,31 @@ type HTTPSConfig struct {
 type RemoteUpdater interface {
 	// UpdateEverything updates all branches, tags, etc., to match the
 	// default remote repository. The implementation is VCS-dependent.
-	UpdateEverything(RemoteOpts) error
+	// If supported by the implementation, results of update will be returned.
+	UpdateEverything(RemoteOpts) (UpdateResult, error)
+}
+
+// UpdateResult is the result of parsing output of the remote update operation.
+type UpdateResult struct {
+	Changes []Change
+}
+
+// Operation that happened to a branch.
+type Operation uint8
+
+const (
+	// New is a branch that was created.
+	New Operation = iota
+
+	// Updated is a branch that was updated.
+	Updated
+
+	// Deleted is a branch that was deleted.
+	Deleted
```

Comment:
These names are nice and don't conflict now, but I'm concerned that using up an exported identifier like `New` for an update result operation name might be problematic in the future if a func `New` is desired to be added.

Perhaps adding an `Op` suffix to these would alleviate that potential problem.